tmt/cli.py

```diff
@@ -566,9 +566,64 @@ def _lint_class(
     return exit_code
 
 
+def _lint_class_collection(
+        context: Context,
+        klass_collection: Union[type[tmt.base.Tests], None],
+        failed_only: bool,
+        enable_checks: list[str],
+        disable_checks: list[str],
+        enforce_checks: list[str],
+        outcomes: list[tmt.lint.LinterOutcome],
+        **kwargs: Any) -> int:
+    """ Lint a class collection of objects """
+
+    exit_code = 0
+
+    if klass_collection is None:
+        return exit_code
+    
+    # FIXME: Workaround https://github.com/pallets/click/pull/1840 for click 7
+    context.params.update(**kwargs)
+
+    linters = klass_collection.resolve_enabled_linters(
+        enable_checks=enable_checks or None,
+        disable_checks=disable_checks or None)
+    
+    lintable = klass_collection.from_tree(context.obj.tree)
+    valid, rulings = lintable.lint(
+        linters=linters,
+        enforce_checks=enforce_checks or None)
+
+    # If the object pass the checks, and we're asked to show only the failed
+    # ones, display nothing.
+    if valid and failed_only:
+        return exit_code
+
+    # Find out what rulings were allowed by user. By default, it's all, but
+    # user might be interested in "warn" only, for example. Reduce the list
+    # of rulings, and if we end up with an empty list *and* user constrained
+    # us to just a subset of rulings, display nothing.
+    allowed_rulings = list(tmt.lint.filter_allowed_checks(rulings, outcomes=outcomes))
+
+    if not allowed_rulings and outcomes:
+        return exit_code
+
+    lintable.print_header()
+
+    echo('\n'.join(tmt.lint.format_rulings(allowed_rulings)))
+
+    if not valid:
+        exit_code = 1
+
+    echo()
+
+    return exit_code
+
+
 def do_lint(
         context: Context,
         klasses: list[Union[type[tmt.base.Test], type[tmt.base.Plan], type[tmt.base.Story]]],
+        klass_lists: list[Union[type[tmt.base.Tests], None]],
```

Comment:
Is a new parameter needed? What's preventing adding this class to `klasses`?

Reply:
I refactored it a bit and was able to remove this additional parameter. Basically, the collection will use the type in `klasses` to determine what the collection to get from the tree.